File: ts/src/209.ts
//3396. Minimum Number of Operations to Make Elements in Array Distinct
function minimumOperations(nums: number[]): number {
  let ops = 0;
  let start = 0;

  while (start < nums.length) {
    const seen = new Set<number>();
    let hasDuplicates = false;

    for (let i = start; i < nums.length; i++) {
      if (seen.has(nums[i])) {
        hasDuplicates = true;
        break;
      }
      seen.add(nums[i]);
    }

    if (!hasDuplicates) break;

    start += 3;
    ops++;
  }

  return ops;
}
//sub: https://leetcode.com/problems/minimum-number-of-operations-to-make-elements-in-array-distinct/submissions/1600177839/?envType=daily-question&envId=2025-04-08
//sol: https://leetcode.com/problems/minimum-number-of-operations-to-make-elements-in-array-distinct/solutions/6627455/3396-minimum-number-of-operations-to-mak-4fwt/

File: rust/src/209.rs
//209. Minimum Size Subarray Sum
impl Solution {
    pub fn min_sub_array_len(target: i32, nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut min_len = i32::MAX;
        let mut left = 0;
        let mut sum = 0;

        for right in 0..n {
            sum += nums[right];

            while sum >= target {
                min_len = min_len.min((right - left + 1) as i32);
                sum -= nums[left];
                left += 1;
            }
        }

        if min_len == i32::MAX {
            0
        } else {
            min_len
        }
    }
}

//sub: https://leetcode.com/problems/minimum-size-subarray-sum/submissions/1604128890/?envType=study-plan-v2&envId=top-interview-150
//sol: https://leetcode.com/problems/minimum-size-subarray-sum/solutions/6641473/209-minimum-size-subarray-sum-by-nalinda-nf30/
